### PR TITLE
Automate frontend linting

### PR DESCRIPTION
**Pull request by asanchezr**
- Add missing enzyme dependencies to package.json
- Add dev dependencies to automate linting on each commit
- Fix linting and formatting errors/warnings

### CODE REVIEW

**Comment by asanchezr** [comment]
This is what I was getting every time I opened the frontend project on VSCode.
Linting is based on default rules from Create-React-App, if you disagree with a particular rule or formatting, let's discuss and disable. For now I didn't customize any rules...

image

**Comment by devinleighsmith** [comment]
Can you annotate what I should be reviewing? I don't need to review the linter's changes.

File: frontend/package.json [hunk 1]
```diff
@@ -33,20 +33,26 @@
     "@testing-library/jest-dom": "4.2.4",
     "@testing-library/react": "9.3.3",
     "@testing-library/user-event": "7.1.2",
+    "@types/enzyme": "3.10.5",
+    "@types/enzyme-adapter-react-16": "1.0.6",
```
Comment (asanchezr) [inline frontend/package.json]: Adding missing enzyme dependencies

File: frontend/package.json [hunk 2]
```diff
@@ -33,20 +33,26 @@
...
+    "@types/react-test-renderer": "16.9.2",
     "@types/redux-logger": "3.0.7",
+    "@types/redux-mock-store": "1.0.2",
     "axios-mock-adapter": "1.17.0",
+    "enzyme": "3.11.0",
+    "enzyme-adapter-react-16": "1.15.2",
     "eslint-config-prettier": "6.7.0",
     "eslint-plugin-prettier": "3.1.1",
+    "husky": "4.2.3",
+    "lint-staged": "10.0.8",
```
Comment (asanchezr) [inline frontend/package.json]: Please review: 
- [husky](https://www.npmjs.com/package/husky) -> allows to add Git hooks without fudging with the .git folder
-  [lint-staged](https://www.npmjs.com/package/lint-staged) -> Run linters against staged git files (before you commit them)

More on why I added them below...

File: frontend/package.json [hunk 3]
```diff
@@ -56,7 +62,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "tsc --noEmit && eslint src --ext ts,tsx,js,jsx --quiet",
+    "lint": "tsc --noEmit && eslint src --quiet --ext .js,.jsx,.ts,.tsx",
```
Comment (asanchezr) [inline frontend/package.json]: just a tweak of the command line args

File: frontend/package.json (hunk 1 above)
Comment (jdtoombs) [reply frontend/package.json]: Woops, forgot to include that on my last PR. Thanks!

File: frontend/package.json [hunk 4]
```diff
@@ -68,6 +74,21 @@
...
+  "lint-staged": {
+    "*.+(js|jsx|ts|tsx)": [
+      "eslint --fix",
+      "git add"
+    ],
+    "*.+(json|css|md)": [
+      "prettier --write",
+      "git add"
+    ]
+  },
```
Comment (asanchezr) [inline frontend/package.json]: This is the main automation piece:
1. Husky will run **lint-staged** on every file to be committed
1. Lint-staged will lint all javascript/typescript files (.js and .ts) that are staged prior to their commit

The above is an optimization of just using Husky and runnint ESLint on the whole code-base on each commit. That can get slow as the app grows. So with lint-staged we only check the new files. 

That is assuming your existing code-base is already linted. That's why I also submitted the lint changes for prior files with this PR.

See:
https://medium.com/@okonetchnikov/make-linting-great-again-f3890e1ad6b8
https://thomlom.dev/setup-eslint-prettier-react/

File: frontend/src/actionCreators/parcelsActionCreator.spec.tsx [hunk 5]
```diff
@@ -37,8 +33,15 @@ describe("fetchParcels action creator", () => {
...
-  it("Request successful, dispatches `success` with correct response", () => {
-    const params: IParcelListParams = { neLat: 1, neLong: 2, swLat: 3, swLong: 4 };
+  it('Request successful, dispatches `success` with correct response', () => {
+    const params: IParcelListParams = {
+      neLat: 1,
+      neLong: 2,
+      swLat: 3,
+      swLong: 4,
+      agencyId: null,
+      propertyClassificationId: null,
```
Comment (asanchezr) [inline frontend/src/actionCreators/parcelsActionCreator.spec.tsx]: To review: I had to add to new properties that were added after this test was written

File: frontend/src/components/common/LookupCodeDropdown.tsx [hunk 6]
```diff
@@ -24,26 +24,28 @@ function LookupCodeDropdown(props: LookupCodeDropdownProps) {
...
+  };
 
   return (
-    <DropdownButton id="dropdown-basic-button" title={codeDropdownTitle} bsPrefix="map-filter-dropdown" onSelect={onCodeItemChange}>
+    <DropdownButton
+      id="dropdown-basic-button"
+      title={codeDropdownTitle}
+      bsPrefix="map-filter-dropdown"
+      onSelect={onCodeItemChange}
+    >
```
Comment (asanchezr) [inline frontend/src/components/common/LookupCodeDropdown.tsx]: To review: The above multi-line break is "enforced" by the max-length of lines we set in **prettier.rc** -> `printWidth: 100,`

If we increase that number we can get it to one-line. How does the team feels about 100 chars. Too long? Too short? I know some people have opinions on how readable is code when it is too long but am happy to adapt to whatever the team wants

File: frontend/src/components/maps/ParcelPopupView.tsx [hunk 7]
```diff
@@ -12,29 +12,47 @@ export interface IParcelDetailProps {
...
+                <Label>Classification: </Label>
+                {parcelDetail?.classification}
+              </ListGroup.Item>
+            </ListGroup>
+          </Col>
+        </Row>
+      )}
+    </Container>
+  );
+};
```
Comment (asanchezr) [inline frontend/src/components/maps/ParcelPopupView.tsx]: The above is also based on rules about using parentheses to wrap components being returned in a function. Again, we are using default values of linting here. We can change/tweak specific rules if we don't like them

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 8]
```diff
@@ -74,28 +74,43 @@ const Map: React.FC<MapProps> = props => {
...
+      <Row style={{ width: '100%', position: 'absolute' }}>
         <MapNavBar />
-        <MapFilterBar agencyLookupCodes={props.agencies} propertyClassifications={props.propertyClassifications} onSelectAgency={handleAgencyChanged} onSelectPropertyClassification={handlePropertyClassificationChanged} />
+        <MapFilterBar
+          agencyLookupCodes={props.agencies}
+          propertyClassifications={props.propertyClassifications}
+          onSelectAgency={handleAgencyChanged}
+          onSelectPropertyClassification={handlePropertyClassificationChanged}
+        />
         <LeafletMap
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/Map.tsx]: More wrapping of long lines

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 9]
```diff
@@ -74,28 +74,43 @@ const Map: React.FC<MapProps> = props => {
...
+          {parcels &&
+            parcels.map(parcel => {
+              return (
+                <Marker
+                  key={parcel.id}
+                  position={[parcel.latitude, parcel.longitude]}
+                  onClick={() => handleParcelClick(parcel)}
+                />
+              );
+            })}
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/Map.tsx]: To review: another rule that the React team feels improves readability (up for discussion)

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx [hunk 10]
```diff
@@ -3,85 +3,129 @@ import renderer from 'react-test-renderer';
...
+    line2: 'N/A',
+    city: 'Victoria',
+    province: 'BC',
+    postal: 'V1V1V1',
+  },
+  landArea: 'unknown',
+  landLegalDescription: 'test',
+  buildings: [],
+  agency: 'FIN',
+};
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx]: Review: This object was using old properties no longer present in **IParcelDetail**

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx [hunk 11]
```diff
@@ -3,85 +3,129 @@ import renderer from 'react-test-renderer';
...
+it('Renders the marker in correct position', () => {
+  const component = mount(
+    <Map
+      lat={48.43}
+      lng={-123.37}
+      zoom={14}
+      parcels={mockParcels}
+      activeParcel={mockDetails}
+      agencies={[]}
+      propertyClassifications={[]}
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx]: I had to add **agencies** and **propertyClassifications** to the unit tests to be able to compile

File: frontend/src/components/navigation/Footer.tsx [hunk 12]
```diff
@@ -3,16 +3,17 @@ import React from 'react';
...
         <Row className="content">
-          <Col xs="auto" md={2}/>
+          <Col xs="auto" md={2} />
           <Col className="navs" md={10}>
             <Nav.Link href="http://www.gov.bc.ca/gov/content/home/disclaimer">Disclaimer</Nav.Link>
             <Nav.Link href="http://www.gov.bc.ca/gov/content/home/privacy">Privacy</Nav.Link>
-            <Nav.Link href="http://www.gov.bc.ca/gov/content/home/accessible-government">Accessibility</Nav.Link>
+            <Nav.Link href="http://www.gov.bc.ca/gov/content/home/accessible-government">
+              Accessibility
+            </Nav.Link>
```
Comment (asanchezr) [inline frontend/src/components/navigation/Footer.tsx]: This one probably went over the 100 chars limit

File: frontend/src/components/navigation/Header.test.tsx [hunk 13]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
 import { render, fireEvent } from '@testing-library/react';
 import renderer from 'react-test-renderer';
-
-jest.mock('@react-keycloak/web');
 import { useKeycloak } from '@react-keycloak/web';
 import Header from './Header';
 
+jest.mock('@react-keycloak/web');
+
```
Comment (asanchezr) [inline frontend/src/components/navigation/Header.test.tsx]: I didn't do this. I have no idea if it impacts the tests...

File: frontend/src/pages/MapView.tsx [hunk 14]
```diff
@@ -19,18 +19,28 @@ const parcelBounds: IParcelListParams = {
...
+  );
+  const lookupCodes = useSelector<RootState, ILookupCode[]>(
+    state => (state.lookupCode as ILookupCodeState).lookupCodes,
+  );
+  const agencies = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.AGENCY_CODE_SET_NAME;
+  });
+  const propertyClassifications = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.PROPERTY_CLASSIFICATION_CODE_SET_NAME;
+  });
```
Comment (asanchezr) [inline frontend/src/pages/MapView.tsx]: More auto-formatting rules that we can discuss

File: frontend/src/components/common/LookupCodeDropdown.tsx (hunk 6 above)
Comment (devinleighsmith) [reply frontend/src/components/common/LookupCodeDropdown.tsx]: starting with 100 seems fine.

File: frontend/src/components/navigation/Header.test.tsx (hunk 13 above)
Comment (devinleighsmith) [reply frontend/src/components/navigation/Header.test.tsx]: I don't think the order matters here.

**Pull request merged**